avocado/core/nrunner.py

```diff
@@ -268,6 +294,8 @@ def pick_runner_class(self, runners_registry=None):
             runners_registry = RUNNERS_REGISTRY_PYTHON_CLASS
 
         runner = runners_registry.get(self.kind, None)
+        if runner is None:
+            runner = self.pick_runner_class_from_entry_point()
```

Comment:
What if we move lines like 296 inside the `pick_runner_class_from_entry_point` and rename the method to something broader like `pick_runner_class`. We can reduce the complexity spread abroad the code and concentrate the task of _pick a runner_ to one method.

Reply:
I like the approach suggested by willian.